bookkeeper-server/src/main/java/org/apache/bookkeeper/tls/BookieAuthZFactory.java

```diff
@@ -95,6 +95,7 @@ public void onProtocolUpgrade() {
                         for (String allowedRole : allowedRoles) {
                             if (certRole[0].equals(allowedRole)) {
                                 authorized = true;
+                                break;
                             }
                         }
                         if (authorized) {
```

Comment:
why do you the `break` here?
it bypasses the rest of the code that is executed depending on the value of `authorized`

Reply:
When we first judge that `authorized` is true, we should break out of the loop to avoid the next useless loop. :)

Reply:
gosh, mentally misaligned closing `}`, makes sense.